tennis-scheduler-web-app/src/components/TennisCourtAddChange.jsx

```diff
@@ -0,0 +1,162 @@
+
+    const handleChange = e => {
+        const { name, value } = e.target;
+        setTennisCourt((tennisCourt) => ({ ...tennisCourt, [name]: value }));
+    };
+
+    const handleChangeAddress = e => {
+        const { name, value } = e.target;
+        setAddress((address) => ({ ...address, [name]: value }));
+    };
+
+    const setImages = e => {
+        const im = e.target.value.split("fakepath")[1].substring(1);
+        setImage(im)
+        setTennisCourt( tennisCourt => ({
+            ...tennisCourt,
+            image: im
+        }))
+        setCheck(true)
+    }
+
+    const onSubmit = e => {
+        e.preventDefault()
+        setTennisCourt( tennisCourt => ({
+            ...tennisCourt,
+            address: address
+        }))
+        setTennisCourt( tennisCourt => ({
+            ...tennisCourt,
+            surfaceType: surface
+        }))
+        setFormErrors(validation(tennisCourt))
+        if (Object.keys(formErrors).length === 0 && !emptyForm) {
+            if (id)
+                tennisCourtChange();
+            else
+                tennisCourtAdd();
+        }
+    }
+
+    const tennisCourtChange = () => {
+        changeTennisCourt(tennisCourt).then(() => {
+            navigate('/');
+            window.location.reload()
+        })
+    }
+
+    const tennisCourtAdd = () => {
+        addTennisCourt(tennisCourt).then(() => {
+            navigate('/');
+            window.location.reload()
+        })
+    }
+
+    const validation = (court) => {
+        const errors = {};
+        const lettersRegex = new RegExp(/^$|^[A-Za-z ]+$/);
+        const streetNumberRegex = new RegExp("^\\d+$");
+        if (!court.name) errors.name = "Name is required!";
+        if (!court.description) errors.description = "Description is required";
+        if (!court.surfaceType) errors.surfaceType = "You have to select surface type!";
+        if (!lettersRegex.test(court.address.country)) errors.country = "Only letters are allowed!"
+        if (!lettersRegex.test(court.address.city)) errors.city = "Only letters are allowed!"
+        if (!lettersRegex.test(court.address.street)) errors.street = "Only letters are allowed!"
+        if (!streetNumberRegex.test(court.address.number)) errors.number = "Only digits are allowed!"
+        setEmptyForm(false);
+        return errors;
+    };
+
+    const handleChangeSurfaceType = e => {
+        setSurface(e.target.value)
+        setChangedType(e.target.value)
+    }
+
+    const onCancel = e => navigate('/')
+
+    useEffect(() => {
+        if (id)
+            getTennisCourt(id).then(
+                response => {
+                    setTennisCourt(response.data)
+                    setAddress(response.data.address)
+                    setCheck(true)
+                    setImage(response.data.image)
+                    setChangedType(response.data.surfaceType)
+                    setSurface(response.data.surfaceType)
+                }
```

Comment:
And also here.

Reply:
It gets data for existing tennis court if it is change that is needed (if path has id in it).

Reply:
What if there is id that don't exist and response is not found? Add catch block and error message for user